code/game/machinery/bots/cleanbot.dm

```diff
@@ -271,6 +271,11 @@ text("<A href='?src=\ref[src];operation=oddbutton'>[src.oddbutton ? "Yes" : "No"
 			armed++
 			icon_state = "roombot_battle[on]"
 			icon_initial = "roombot_battle"
+	else if (istype(W, /obj/item/clothing/head/maidhat))
+		var/obj/machinery/bot/cleanbot/roomba/meido/M = new(get_turf(src))
+		M.name = name
+		to_chat(user, "<span class='notice'>You attach \the [W] to \the [src]. It makes a soft booping noise and its light blinks excitedly for a moment.</span>")
+		qdel(src)
```

Comment:
If there's a pAI in the roomba, will this delete it?

Reply:
Yes, nice catch.